### PR TITLE
fix: job state shows as scheduled when resources are allocated

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-166]

## Description

Kubernetes job status on the Cluster Page will correctly reflect the state of workloads. Once all Kubernetes pods are scheduled on nodes, the job state will be reflected as "scheduled" rather than "queued".

## Test Plan

None needed.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RM-166]: https://hpe-aiatscale.atlassian.net/browse/RM-166?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/rm/kubernetesrm/job.go [hunk 1]
```diff
@@ -587,6 +587,16 @@ func (j *job) preparePodUpdateMessage(msgText string) string {
 	return msgText
 }
 
+// PodScheduled checks pod conditions to determine if a pod has been scheduled onto a node.
+func PodScheduled(pod k8sV1.Pod) bool {
```
Comment (stoksc) [inline master/internal/rm/kubernetesrm/job.go:591]: style nit, don't export this function since its usefulness is limited to the k8s package

**Review by stoksc — APPROVED** [review]
lgtm, nice!

**Comment by stoksc** [comment]
also, a test would be nice. should've said that first time around, oops.

File: docs/release-notes/job-state.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**Bug Fixes**
+
+-  Kubernetes: Job state will now correctly show as "SCHEDULED" once all pods have been assigned to
+   nodes. Previously, jobs would remain in state "QUEUED" until all pods were in phase "Running".
```
Comment (tara-det-ai) [inline docs/release-notes/job-state.rst:6]: :orphan:

**Bug Fixes**

-  Kubernetes: Fix an issue where where jobs would remain in "QUEUED" state until all pods were running. Jobs will now correctly show as "SCHEDULED" once all pods have been assigned to nodes.

**Review by tara-det-ai — APPROVED** [review]
requested changes

File: docs/release-notes/job-state.rst (hunk 2 above)
Comment (kkunapuli) [reply docs/release-notes/job-state.rst]: Thanks! Yes, this is much better phrasing. :)

**Comment by kkunapuli** [comment]
> also, a test would be nice. should've said that first time around, oops.

Great idea - added a test!

File: master/internal/rm/kubernetesrm/jobs_test.go [hunk 3]
```diff
@@ -63,18 +64,74 @@ func TestGetNonDetPods(t *testing.T) {
...
 	}
 
-	actualPods, err := p.getNonDetPods()
+	actualPods, err := js.getNonDetPods()
 	require.NoError(t, err)
 	require.ElementsMatch(t, expectedPods, actualPods)
 }
 
+func TestJobScheduledStatus(t *testing.T) {
+	pendingPod := k8sV1.Pod{
```
Comment (stoksc) [inline master/internal/rm/kubernetesrm/jobs_test.go:80]: i usually prefer declaring these closer to where they are used but with such a simple test i dont think it affects readability

**Review by stoksc — APPROVED** [review]

**Pull request merged**